.circleci/config.yml

```diff
@@ -0,0 +1,296 @@
+version: 2.0
+
+aliases:
+  # Workspace
+  - &workspace
+    ~/video-quickstart-android
+
+  - &gradle-cache-key
+    key: jars-{{ checksum "build.gradle" }}-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}
+  - &restore-cache-gradle
+    <<: *gradle-cache-key
+    name: Restore Gradle Cache
+  - &save-cache-gradle
+    <<: *gradle-cache-key
+    name: Save Gradle Cache
+    paths:
+      - ~/.gradle/caches
+      - ~/.gradle/wrapper
+
+  # Configure git user
+  - &configure-git-user
+    name: Configure git user
+    command: |
+      git config --global user.email "twilio-sdk-build@twilio.com"
+      git config --global user.name "twilio-sdk-build"
+
+  # Filters
+  - &filter-snapshot
+    branches:
+      only:
+        - master
+
+  # Containers
+  - &build-defaults
+    working_directory: *workspace
+    docker:
+      - image: circleci/android:api-28-node8-alpha
+    environment:
+      - _JAVA_OPTIONS: "-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap"
+
+jobs:
+  setup-workspace:
+    <<: *build-defaults
+    resource_class: medium+
+    steps:
+      # Setup code and workspace for downstream jobs
+      - checkout
+      - restore-cache: *restore-cache-gradle
+      
+      # Save cache
+      - save-cache: *save-cache-gradle
+
+  check-format:
+    <<: *build-defaults
+    resource_class: medium+
+    steps:
+      # Setup
+      - checkout
+      - restore-cache: *restore-cache-gradle
+
+      # Run spotless check
+      - run:
+          name: Spotless Check
+          command: ./gradlew -q spotlessCheck
+
+      # Save cache
+      - save-cache: *save-cache-gradle
+
+  build-exampleAdvancedCameraCapturer:
+    <<: *build-defaults
+    resource_class: xlarge
+    steps:
+      # Setup
+      - checkout
+      - attach_workspace:
+          at: *workspace
+      - restore-cache: *restore-cache-gradle
+
+      # Build app
+      - run:
+          name: Build exampleAdvancedCameraCapturer
+          command: ./gradlew -q exampleAdvancedCameraCapturer:assemble
+
+      # Save the builds to workspace
```

Comment:
Persisting to workspaces is not required. There are no downstream jobs using the result of these builds.